### ISSUE TITLE
ReactNative:  Error: Creating URL objects is not supported yet.

### ISSUE THREAD

**Issue by soulfly** [xmppjs/xmpp.js#547]
This example 

`client.start('ws://localhost:5280/xmpp-websocket')`

not working under ReactNative, I'm getting the following error:

```
16:25:17: [Unhandled promise rejection: Error: Creating URL objects is not supported yet.]
- node_modules/react-native/Libraries/Blob/URL.js:53:20 in URL
* xmpp.js/packages/connection/index.js:51:22 in getDomain
* xmpp.js/packages/connection/index.js:228:33 in start$
- node_modules/regenerator-runtime/runtime.js:62:44 in tryCatch
- node_modules/regenerator-runtime/runtime.js:296:30 in invoke
- node_modules/regenerator-runtime/runtime.js:62:44 in tryCatch
- node_modules/regenerator-runtime/runtime.js:152:28 in invoke
- node_modules/regenerator-runtime/runtime.js:195:17 in <unknown>
- node_modules/promise/setimmediate/core.js:45:7 in tryCallTwo
- node_modules/promise/setimmediate/core.js:200:23 in doResolve
- node_modules/promise/setimmediate/core.js:66:12 in Promise
- node_modules/regenerator-runtime/runtime.js:194:27 in callInvokeWithMethodAndArg
- node_modules/regenerator-runtime/runtime.js:217:12 in enqueue
- node_modules/regenerator-runtime/runtime.js:241:8 in async
```

I use the latest master branch

What we need to do is to specify a `domain` manually, like this:

```
var options =  {
  "domain": "some.chat.server.com",
  "uri": "ws://some.chat.server.com:5280/xmpp-websocket",
}
client.start(options)
```

As I understand, it can't be fixed now because of lack of RN support https://github.com/facebook/react-native/issues/16434, 
could you please update documentation with this thing?

**Comment by sonnyp** [comment]
interesting, so NativeScript uses React Native?

It can be fixed by requiring some polyfill or `url` instead of assuming global `URL` is defined. Not sure yes what's the best approach. Maybe document you need a polyfill if you environment doesn't support that feature.

**Comment by soulfly** [comment]
Sorry, my mistake, it's about ReactNative..

**Comment by sonnyp** [comment]
Fixed in 0.5.1 https://github.com/xmppjs/xmpp.js/releases/tag/v0.5.1
